cmd/iam.go

```diff
@@ -312,11 +312,13 @@ func (sys *IAMSys) Init(ctx context.Context, objAPI ObjectLayer, etcdClient *etc
 	switch {
 	case globalOpenIDConfig.ProviderEnabled():
 		go func() {
-			ticker := time.NewTicker(sys.iamRefreshInterval)
-			defer ticker.Stop()
+			timer := time.NewTimer(sys.iamRefreshInterval)
+			defer timer.Stop()
 			for {
 				select {
-				case <-ticker.C:
+				case <-timer.C:
+					timer.Reset(sys.iamRefreshInterval)
```

Comment:
Shouldn't this be at the end of the this case block so that the timer is started after the work for this interval is done? Otherwise it doesn't seem to be different from using ticker.

Reply:
No timer won't work on it until there is timer.C @donatello

Reply:
timer.Reset() doesn't mean it starts counting, it simply resets the counter to said value - the caller has to still do `<-timer.C` to start counting.

Reply:
That does not seem to be the case - https://go.dev/play/p/qoF71_D1oXD

If we have a job that lasts 2 seconds and the timer interval is for a second, we want it to execute only every 3 seconds as show here. If we call Reset as soon as we drain the channel, the job get run every 2 seconds.